Kudu.Core/Tracing/XmlTracer.cs

```diff
@@ -175,6 +187,24 @@ private void WriteEndTrace()
             }
         }
 
+        public static bool ShouldTrace(TraceLevel level, TimeSpan elapsed, string file)
+        {
+            // if SCM_TRACE_LEVEL is set to 4 (verbose), always trace.
+            if (level >= TraceLevel.Verbose)
+            {
+                return true;
+            }
+
+            // for api taking long time (>10s), always trace.
+            if (elapsed >= ElapsedThreshold)
+            {
+                return true;
+            }
+
+            // filter out those that may be polled by portal.
+            return !TraceFilters.Any(filter => file.IndexOf(filter, StringComparison.OrdinalIgnoreCase) > 0);
```

Comment:
Seems a bit fragile to be filtering based on the complex file naming pattern. e.g. if we decide to tweak our file names, all the filters would need to be updated.